src/python/WMComponent/RucioInjector/Database/MySQL/GetCompletedBlocks.py

```diff
@@ -0,0 +1,75 @@
+"""
+_GetCompletedBlocks_
+
+MySQL implementation of RucioInjector.GetClompletedBlocks
+
+Retrieve a list of blocks that are complete,
+including their location and the sites they
+are subscribed too
+
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+from WMCore.Database.DBFormatter import DBFormatter
+
+
+class GetCompletedBlocks(DBFormatter):
+    # Retrieves a list of blocks that are closed but sure if deletedable:
+    #   - workflow completions for all files in the block to be checked on a later stage
+    #   - subscription made for dataset is copy+delete
+    #   - subscription has been made in PhEDEx
+    #   - blocks hasn't been deleted yet
+
+    sql = """SELECT dbsbuffer_block.blockname,
+                    dbsbuffer_location.pnn,
+                    dbsbuffer_dataset.path,
+                    dbsbuffer_dataset_subscription.site,
+                    dbsbuffer_workflow.name
+             FROM dbsbuffer_dataset_subscription
+             INNER JOIN dbsbuffer_dataset ON
+               dbsbuffer_dataset.id = dbsbuffer_dataset_subscription.dataset_id
+             INNER JOIN dbsbuffer_block ON
+               dbsbuffer_block.dataset_id = dbsbuffer_dataset_subscription.dataset_id
+             INNER JOIN dbsbuffer_file ON
+               dbsbuffer_file.block_id = dbsbuffer_block.id
+             INNER JOIN dbsbuffer_workflow ON
+               dbsbuffer_workflow.id = dbsbuffer_file.workflow
+             INNER JOIN dbsbuffer_location ON
+               dbsbuffer_location.id = dbsbuffer_block.location
+             WHERE dbsbuffer_dataset_subscription.delete_blocks = 1
+             AND dbsbuffer_dataset_subscription.subscribed = 1
+             AND dbsbuffer_block.status = 'Closed'
+             AND dbsbuffer_block.deleted = 0
```

Comment:
We need to add a configuration parameter for RucioInjector (`deleteBlocksDelay` ?) to control for how long we will allow the data to remain on disk.  We need this so we can keep de data for a few days, in case there is an urgent need for it. Can you modify this query to use such a parameter to filter the blocks according to  `dbsbuffer_block.create_time`?

Reply:
thanks @germanfgv Good point.

Reply:
fixed as well.